dali/pipeline/operators/reader/loader/video_loader.cc

```diff
@@ -356,18 +393,38 @@ void VideoLoader::read_file() {
           continue;
       }
 
-      auto frame = av_rescale_q(pkt->pts,
+      auto frame = av_rescale_q(pkt->pts - file.start_time_,
                                 file.stream_base_,
                                 file.frame_base_);
       LOG_LINE << "Frame candidate " << frame << " (for " << req.frame  <<" )...\n";
 
       file.last_frame_ = frame;
-      auto key = pkt->flags & AV_PKT_FLAG_KEY;
+      key = (pkt->flags & AV_PKT_FLAG_KEY) != 0;
+      int pkt_frames = 1;
+      if (pkt->duration) {
```

Comment:
```suggestion
      if (pkt->duration > 0) {
```
should it be that?

Reply:
No. Zero indicates "not set" - then I assume packet is one frame.